ntoskrnl/mm/ARM3/wslist.cpp

```diff
@@ -293,7 +294,7 @@ TrimWsList(PMMWSL WsList)
 
             /* Make this a transition PTE */
             MI_MAKE_TRANSITION_PTE(PointerPte, Page, Protection);
-            KeInvalidateTlbEntry(MiAddressToPte(PointerPte));
+            KeFlushSingleTb(MiAddressToPte(PointerPte), FALSE);
```

Comment:
Noob question (I could have asked it in other places): What dictates the choice of flushing the TLB only for the processors on which the "current" process is running, versus, flushing unconditionally on all processors?

Reply:
If it's a process local page, then there is no need to flush it on processors that don't run in the context of the current thread, because they first need to switch to it, i.e. switch cr3, which flushes the TLB already for non-global pages. And WSLE entries are process local.